decidim-core/lib/decidim/geocodable.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "geocoder/stores/active_record"
+
+module Decidim
+  # This concern overrides some of the Active Record functionality injected by
+  # the Geocoder gem in order to pass the geocoding searches through Decidim's
+  # own API which configures the geocoder correctly for each search. This is
+  # used when the `model.geocode` and `model.reverse_geocode` methods are called
+  # for the Active Record models.
+  #
```

Comment:
Every model in Decidim that uses either `geocoded_by` or `reverse_geocoded_by` is affected by this. It turns into action when you call `model.geocode` or `model.reverse_geocode` as explained by that comment.

This is needed in order to configure the Geocoder for each individual geocoding request according to the Decidim API.

It basically overrides the `do_lookup` method from this module:
https://github.com/alexreisner/geocoder/blob/353a434e9628df5dcf8d365708aed6230833c999/lib/geocoder/stores/base.rb

The `RecordGeocoder` is called by this line as we override the `Geocoder` constant for that module:
https://github.com/alexreisner/geocoder/blob/353a434e9628df5dcf8d365708aed6230833c999/lib/geocoder/stores/base.rb#L100

This allows us to keep the record context in the geocoder queries in order to pass the geocoder queries through the Decidim API. The record context is relevant because the maps utilities are context aware and need the organization information.

If, for some reason, the geocoded model is not aware of its organization, it will fall back to the default functionality in the Geocoder gem.